runners/direct-java/src/main/java/org/apache/beam/runners/direct/SideInputContainer.java

```diff
@@ -169,7 +170,13 @@ public CallbackSchedulingLoader(
         load(PCollectionViewWindow<?> view) {
 
       AtomicReference<Iterable<? extends WindowedValue<?>>> contents = new AtomicReference<>();
-      WindowingStrategy<?, ?> windowingStrategy = view.getView().getWindowingStrategyInternal();
+      checkArgument(
+          view.getView() instanceof SimplePCollectionView,
+          "Unknown %s type: %s",
```

Comment:
Considering the number of occurrences, suggest factoring out the `checkState` and downcast into something marked `@Internal` in `util/`. Especially since there is no actual concept that we would ever support any other kind of view, we can make the error message extremely explicit: don't make other views